py/server/deephaven/experimental/iceberg.py

```diff
@@ -0,0 +1,265 @@
+    instructions and table definitions, as well as special data instructions for loading data files from the cloud.
+    """
+
+    j_object_type = _JIcebergInstructions or type(None)
+
+    def __init__(self,
+                 table_definition: Optional[Union[Dict[str, DType], List[Column]]] = None,
+                 data_instructions: Optional[s3.S3Instructions] = None,
+                 column_renames: Optional[Dict[str, str]] = None):
+
+        """
+        Initializes the instructions using the provided parameters.
+
+        Args:
+            table_definition (Union[Dict[str, DType], List[Column], None]): the table definition; if ommitted, the
+                definition is inferred from the Iceberg schema. Setting a definition guarantees the returned table will
+                have that definition. This is useful for specifying a subset of the Iceberg schema columns.
+            data_instructions (Optional[s3.S3Instructions]): Special instructions for reading data files, useful when
+                reading files from a non-local file system, like S3.
+            column_renames (Optional[Dict[str, str]]): A dictionary of old to new column names that will be renamed in
+                the output table.
+
+        Raises:
+            DHError: If unable to build the instructions object.
+        """
+
+        if not _JIcebergInstructions:
+            raise DHError(message="IcebergInstructions requires the Iceberg specific deephaven extensions to be "
+                                  "included in the package")
+
+        try:
+            builder = self.j_object_type.builder()
+
+            if table_definition is not None:
+                builder.tableDefinition(_j_table_definition(table_definition))
+
+            if data_instructions is not None:
+                builder.dataInstructions(data_instructions.j_object)
+
+            if column_renames is not None:
+                for old_name, new_name in column_renames.items():
+                    builder.putColumnRenames(old_name, new_name)
+
+            self._j_object = builder.build()
+        except Exception as e:
+            raise DHError(e, "Failed to build Iceberg instructions") from e
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self._j_object
+
+def _j_table_definition(table_definition: Union[Dict[str, DType], List[Column], None]) -> Optional[jpy.JType]:
+    if table_definition is None:
+        return None
+    elif isinstance(table_definition, Dict):
+        return _JTableDefinition.of(
+            [
+                Column(name=name, data_type=dtype).j_column_definition
+                for name, dtype in table_definition.items()
+            ]
+        )
+    elif isinstance(table_definition, List):
+        return _JTableDefinition.of(
+            [col.j_column_definition for col in table_definition]
+        )
+    else:
+        raise DHError(f"Unexpected table_definition type: {type(table_definition)}")
```

Comment:
use the one in `parquet.py` or better yet move it to `jcompat.py` as an util func?